### ISSUE TITLE
_ or chain function for Lo-Dash FP

### ISSUE THREAD

**Issue by yannickglt** [lodash/lodash#3692]
Hey!

What do you think of such a `_` or `chain` function which could be exported by Lo-Dash FP:
```js
const _ = (val) => ({ pipe: (...flow) => pipe(...flow)(val)  });
```
It could improve readability for users used to Lo-Dash and to RxJS. It only wraps the pipe function (or the flow one) and would make the sense of reading more natural. 
I already use it as is but I think it might interest other people.

### Before
```js
const values = pipe(
  map(x => [x, x*2]),
  flatten,
  sortBy(x => x)
)([1, 2, 3]);
```

### After
```js
const result = _([1, 2, 3])
 .pipe(   
  map(x => [x, x*2]),
  flatten,
  sortBy(x => x)
 );
```

**Comment by robinpokorny** [comment]
This violates the data-last principle. You can use `_.flow` for the same use case:

```js
const result = _.flow(
  map(x => [x, x * 2]),
  flatten,
  sortBy(x => x)
)([1, 2, 3]);
```

**Comment by robinpokorny** [comment]
When I quickly looked into the `lodash/fp` guide, I discovered there is a section discussing it. The reasons why `chain` is not included are summed in this article https://medium.com/making-internets/why-using-chain-is-a-mistake-9bc1f80d51ba

**Comment by yannickglt** [comment]
Thanks for your answer.
I didn't know that "data-last"  was a good practice and a principle to follow. Good to know, I will try to take the habit. The main disturbing thing I guess is the fact the variable passed as input to the flow may appear several lines after the assignment to the flow output.
I do know this article and I really like it. That's the main reason I moved to Lodash FP. The chain function is not a good solution, as explained in the post. However, composé or flow (or pipe) provide a good way to chain (because yes it is chaining) Lodash functions.
Thanks again for the great work you do for us.

**Comment by trubit** [comment]
I just came across lodash FP to switch from normal - I'm not like all-in for FP, but I want to import only used functions (same as RXJS) and far as I know this is only way to do it and also write it sane way.
I'm using lodash with typescript and this is actually issue for intellisense. 

When writing code you first write implementation but intellisense doesnt know what data type you write in, so you won't get proper hints for 'piped' functions.

I understand data-lasts principle, but in typescript or at least way the typings for lodash/fp are written this doenst help much - and i prefer autocomplete/intellisense over some principle :)

So this is what I came up with:
`
import { pipe, map, sum } from 'lodash/fp';
const data = [1, 2, 3];
pipe(
  () => data,
  map(n => n * 2),
  sum
)();
`

But seems more like hack then solution, maybe suggested solutions could be re-evaulated? Or is there other way?
